### PR TITLE
Enable stock HuggingFace GPT2LMHeadModel on device=nntile

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Makes **stock** `transformers.GPT2LMHeadModel` work with `model.to("nntile")` — no custom `torch_nntile` model wrapper required.

```python
import torch_nntile  # registers device + HF compat patches
from transformers import GPT2Config, GPT2LMHeadModel

config = GPT2Config(..., attn_pdrop=0.0, resid_pdrop=0.0, embd_pdrop=0.0)
config._attn_implementation = "sdpa"
model = GPT2LMHeadModel(config).eval().float().to("nntile")

logits = model(input_ids).logits
loss = torch_nntile.training.cross_entropy(logits, labels)
loss.backward()
```

### New aten kernels (backend gaps for HF GPT-2)
- `sum` — Conv1D / addmm backward reductions
- `mul.Scalar` — activation scaling

### HF compat (`torch_nntile/compat/hf.py`, auto-imported)
- `gelu_new` → `F.gelu(approximate="tanh")` on nntile
- ND `F.linear` reshape workaround (3D linear backward)
- Auto `tie_weights()` after `PreTrainedModel.to("nntile")`
- GPT-2 `cache_position` created on CPU, copied via `.to("nntile")` (no `arange` kernel)
- GPT-2 `wpe` output expanded to batch size before token+position add (no broadcast `add` kernel)

### Training helper fix
- `train_full_batch_step` extracts `.logits` from HF `ModelOutput`

### Tests & example
- `test_hf_gpt2_stock_on_nntile.py` — forward, backward grad parity, train step
- `examples/gpt2_hf_out_of_box.py` — end-to-end demo

## Notes

- Requires `import torch_nntile` before model construction
- Dropout must be 0 and attention must use `sdpa` for parity on CPU-only builds
- ND linear backward still uses a Python reshape shim; proper C++ fix in `nntile_linear.cpp` is a follow-up
- Supersedes the closed minimal-model PR #444

<div><a href="https://cursor.com/agents/bc-019f3206-e89d-7951-82d9-d00c0bb0ecb2"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-019f3206-e89d-7951-82d9-d00c0bb0ecb2"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor review

**Comment by Muxas** [comment]
@cursor review
